linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -2732,6 +2732,7 @@ public static <TSource, TKey> Enumerable<TSource> orderBy(
           // search the key up to (but excluding) which we have to remove entries from the map
           int skipped = 0;
           TKey until = null;
+          boolean hasResult = false;
```

Comment:
I wonder if it would be clearer to keep the same approach in the code, but just using another "unset value" instead of `null`, e.g. the `DUMMY` object that already exists in the class:
```
TKey until = (TKey) DUMMY;
...
if (until == DUMMY)
```